test/unit/core.js

```diff
@@ -687,17 +687,19 @@ QUnit.test( "jQuery('html')", function( assert ) {
 	assert.equal( jQuery( "\\<div\\>" ).length, 0, "Ignore escaped html characters" );
 } );
 
-QUnit.test( "jQuery(tag-hyphenated elements) gh-1987", function( assert ) {
-	assert.expect( 17 );
-
-	jQuery.each( "thead tbody tfoot colgroup caption tr th td".split( " " ), function( i, name ) {
-		var j = jQuery( "<" + name + "-d></" + name + "-d>" );
-		assert.ok( j[ 0 ], "Create a tag-hyphenated elements" );
-		assert.ok( jQuery.nodeName( j[ 0 ], name.toUpperCase() + "-D" ), "Tag-hyphenated element has expected node name" );
+QUnit.test( "jQuery(symbol separated elements)", function( assert ) {
+	assert.expect( 34 );
+
+	jQuery.each( "- :".split( " " ), function( i, symbol ) {
+		jQuery.each( "thead tbody tfoot colgroup caption tr th td".split( " " ), function( k, element ) {
+			var j = jQuery( "<" + element + symbol + "test></" + element + symbol + "test>" );
```

Comment:
And _this_ is getting even harder to read. Let's just introduce a `tagName` variable.